compileopts/target.go

```diff
@@ -308,13 +322,32 @@ func defaultTarget(goos, goarch, triple string) (*TargetSpec, error) {
 	case "arm":
 		spec.CPU = "generic"
 		spec.CFlags = append(spec.CFlags, "-fno-unwind-tables", "-fno-asynchronous-unwind-tables")
+		hardfloat := strings.HasSuffix(triple, "hf")
 		switch strings.Split(triple, "-")[0] {
 		case "armv5":
-			spec.Features = "+armv5t,+strict-align,-aes,-bf16,-d32,-dotprod,-fp-armv8,-fp-armv8d16,-fp-armv8d16sp,-fp-armv8sp,-fp16,-fp16fml,-fp64,-fpregs,-fullfp16,-mve.fp,-neon,-sha2,-thumb-mode,-vfp2,-vfp2sp,-vfp3,-vfp3d16,-vfp3d16sp,-vfp3sp,-vfp4,-vfp4d16,-vfp4d16sp,-vfp4sp"
+			if hardfloat {
+				spec.CFlags = append(spec.CFlags, "-mfloat-abi=hard")
+				spec.Features = "+armv5t,+strict-align,-aes,-bf16,-d32,-dotprod,-fp-armv8,-fp-armv8d16,-fp-armv8d16sp,-fp-armv8sp,-fp16,-fp16fml,-fp64,-fpregs,-fullfp16,-mve.fp,-neon,-sha2,-thumb-mode,-vfp2,-vfp2sp,-vfp3,-vfp3d16,-vfp3d16sp,-vfp3sp,-vfp4,-vfp4d16,-vfp4d16sp,-vfp4sp"
+			} else {
+				spec.CFlags = append(spec.CFlags, "-mfloat-abi=soft") // maybe use softfp instead?
```

Comment:
`softfp` means to still use the FPU (but use an ABI compatible with soft-float code), which is not what we want here. So this comment can be removed.

```suggestion
				spec.CFlags = append(spec.CFlags, "-mfloat-abi=soft")
```

Same goes for the other two similar comments below.